### PR TITLE
Josh fixes on django 110

### PR DESCRIPTION
**Pull request by yoshrote**
there are some url resolving issue that crop up once we hit django 1.10 and 1.11 but otherwise things are looking good.

### CODE REVIEW

**Comment by yoshrote** [comment]
i created a PR to test how these changes affect djserver. https://github.com/handshake/handshake/pull/3250 
`djserver.handshake.tests.api.common.custom_object_resource_tests:CustomObjectResourceTests.test_object_validation` broke from the changes. i need to update that PR to run against more recent changes in this branch and we need to decide if we should figure out alternate tastypie fixes or just update the tests in djserver

**Comment by slinkp** [comment]
no trolls please

File: tests/gis/tests/http.py [hunk 1]
```diff
@@ -2,7 +2,8 @@
 from urllib import quote
 from testcases import TestServerTestCase
 import json
-
+import django
+django18 = django.VERSION[1] <= 8
```
Comment (slinkp) [inline tests/gis/tests/http.py]: 'nother one of these

File: tests/gis/tests/views.py [hunk 2]
```diff
@@ -1,6 +1,8 @@
 from django.http import HttpRequest
 from testcases import TestCaseWithFixture as TestCase
 import json
+import django
+django18 = django.VERSION[1] <= 8
```
Comment (slinkp) [inline tests/gis/tests/views.py]: and here

**Pull request merged**